### PR TITLE
Improve team settings workflows

### PR DESCRIPTION
**Pull request by jakob223**
Key a bunch of stuff on team ID, and generally clean stuff up.

### CODE REVIEW

**Comment by chrisnojima** [comment]
might be nice to change the team waiting key so you don't need to plumb so much state everywhere just to get the name

**Comment by jakob223** [comment]
> might be nice to change the team waiting key so you don't need to plumb so much state everywhere just to get the name

I was planning to change it eventually when we reach a tipping point; do you think it's reached?

**Comment by chrisnojima** [comment]
doesn't have to be a part of this pr but it'll allow you to go back to chainAction etc

File: go/teams/get.go [hunk 1]
```diff
@@ -30,21 +30,21 @@ func GetForTeamManagementByStringName(ctx context.Context, g *libkb.GlobalContex
...
 		ForceRepoll:               true,
 		AllowNameLookupBurstCache: true,
 	})
 	if err != nil {
-		return nil, fixupTeamGetError(ctx, g, err, name, public)
+		return nil, fixupTeamGetError(ctx, g, err, id.String(), public)
 	}
 	if team.IsImplicit() {
-		return nil, fmt.Errorf("cannot display implicit team by name")
+		return nil, fmt.Errorf("cannot display implicit team by ID")
```
Comment (mlsteele) [inline go/teams/get.go]: IsImplicit check doesn't make sense anymore. `GetForDisplayByTeamID` has only one callsite so just delete `GetForDisplayByTeamID` and roll the `Load` into the callsite.

File: go/teams/service_helper.go [hunk 2]
```diff
@@ -1616,6 +1616,31 @@ func ChangeTeamSettings(ctx context.Context, g *libkb.GlobalContext, teamName st
 	})
 }
 
+func ChangeTeamSettingsByID(ctx context.Context, g *libkb.GlobalContext, id keybase1.TeamID,
+	settings keybase1.TeamSettings) error {
```
Comment (mlsteele) [inline go/teams/service_helper.go]: How about making `ChangeTeamSettings` resolve the ID and then pass it to `ChangeTeamSettingsByID`? Also callers of `ChangeTeamSettings` are good candidates for this same ID treatment.

Comment (jakob223) [reply go/teams/service_helper.go]: Changed about half of them, don't really feel like doing the rest, does it feel important to you

**Comment by jakob223** [comment]
Bump @chrisnojima

File: go/teams/service_helper.go (hunk 2 above)
Comment (mlsteele) [reply go/teams/service_helper.go]: Huh maybe I'm looking differently but now `TeamSetSettings` is the only remaining one. Not more important than other things in this PR

**Pull request merged**